src/GraphQL/FromServicesAttribute.cs

```diff
@@ -10,11 +10,8 @@ namespace GraphQL
     public class FromServicesAttribute : GraphQLAttribute
     {
         /// <inheritdoc/>
-        public override void Modify<TParameterType>(ArgumentInformation<TParameterType> argumentInformation)
-            => argumentInformation.Expression = context => GetRequiredService<TParameterType>(context);
-
-        private static TServiceType GetRequiredService<TServiceType>(IResolveFieldContext context)
-            => (context.RequestServices ?? throw new MissingRequestServicesException())
-                .GetRequiredService<TServiceType>();
+        public override void Modify<TParameterType>(ArgumentInformation argumentInformation)
+            => argumentInformation.SetExpression(context => (context.RequestServices ?? throw new MissingRequestServicesException())
+                .GetRequiredService<TParameterType>());
```

Comment:
In order to be able to EASILY set a type-safe expression, without having to write `(Expression<Func<IResolveFieldContext, TParameterType>>)`, I added a generic method called `SetExpression`.  I also made this method accept a delegate, which is honestly more flexible in its use, as shown here where we can use null coalescing operators, code blocks, and so on.

Reply:
> to be able to EASILY set a type-safe expression

OK. I'm also want to end with the most easy solution. Consider `HardcodedValueAttribute` from my latest commit. One wants to return hardcoded value 85. So he should to cast 85 to `TParameterType` , right? Something like `context => (TParameterType)(object)85`. Do you mean that this syntax is simple? Type-safety is ephemeral here. In my variant the code looks like `context => 85`.

Reply:
Exactly -- you are hiding the inherit loss of type safety.  It should be explicit.

Reply:
😕  Then there should be unwrapping code for `Convert` expression in your variant for such cases if you really want to eliminate boxing and achive the same performance as with direct method call.